conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -1353,3 +1353,45 @@ def build(self):
                  "CMakeLists.txt": cmake})
     client.run("create . -pr=myprofile")
     assert "-- MYVAR1 MYVALUE1!!" in client.out
+
+
+@pytest.mark.tool("cmake")
+def test_redirect_stdout():
+    client = TestClient()
+    conanfile = textwrap.dedent("""
+    import os
+    from conan import ConanFile
+    from conan.tools.cmake import CMake, CMakeToolchain
+    from conan.tools.cmake import cmake_layout
+
+    class Pkg(ConanFile):
+        name = "foo"
+        version = "1.0"
+        settings = "os", "arch", "build_type", "compiler"
+
+        exports_sources = "CMakeLists.txt", "main.cpp"
+
+        def layout(self):
+            cmake_layout(self)
+
+        def generate(self):
+            CMakeToolchain(self).generate()
+
+        def build(self):
+            self.output.info(f"Metadata: {self.package_metadata_folder}")
+
```

Comment:
I'd probably try to simplify this test:
- generators = "CMakeToolchain"
- remove output.info()
- no ``layout()`` necessary
- Probably better decouple the test from the metadata feature. Just dump in the build folder. If build goes wrong no metadata is generated